src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -0,0 +1,53 @@
+package roomescape.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import roomescape.dao.ReservationDao;
+import roomescape.domain.Reservation;
+import roomescape.dto.ReservationRequest;
+import roomescape.dto.ReservationResponse;
+import roomescape.entity.ReservationEntity;
+
+import java.net.URI;
+import java.util.List;
+
+@Controller
+@RequestMapping("/reservations")
+public class ReservationController {
+    private final ReservationDao reservationDao;
+
+    @Autowired
```

Comment:
생성자가 하나뿐이기 때문에, 생략해도 자동으로 주입되는 것 같네요!
생략하면 간결해지는 장점은 있지만, 생략하지 않으면 "구현체를 주입받을 것이고 현재 인터페이스에 의존하고 있다"라는 것을 명시하는 느낌이 드네요.

Reply:
https://docs.spring.io/spring-framework/reference/core/beans/dependencies/factory-collaborators.html

> Since you can mix constructor-based and setter-based DI, it is a good rule of thumb to use constructors for mandatory dependencies and setter methods or configuration methods for optional dependencies. Note that use of the [@Autowired](https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html) annotation on a setter method can be used to make the property be a required dependency; however, constructor injection with programmatic validation of arguments is preferable.

참고하시면 좋을 것 같아요 ㅎㅎ